api/routes.py

```diff
@@ -569,6 +569,20 @@ def saml_authenticate():
 def saml_callback():
     return app.manager.saml_controller.saml_authentication_callback(request, app.manager._db)
 
+
+@library_route('/lcp/hint')
+@has_library
+@returns_problem_detail
+def lcp_passphrase():
+    return app.manager.lcp_controller.get_lcp_passphrase()
```

Comment:
I see a couple problems with these new routes:

1. These routes need to use `@requires_auth` since they can't (lcp_passphrase) or shouldn't (lcp_license) be used unauthenticated.
2. How are clients to know about the URLs? I don't see any links to these URLs in this branch; will our license documents include these links? In the LCP spec I see a definition of a link relation `hint` which could link to `lcp_passphrase` but I don't see what would be using `lcp_license` -- maybe a `self` link?

Reply:
1. Internally they use `BaseCirculationManagerController.authenticated_patron_from_request`. I'll add `@requires_auth` 
2. **The hint link** becomes a part of an LCP license:
```
"links": [{
    "hash": "4b82c412664b4b4b2841f4abb9a0cd1f2bd1a98a815026f83db25f010427487f",
    "title": "9780231543972.epub",
    "length": 798385,
    "href": "http://localhost:9000/encrypted-books/9780231543972",
    "rel": "publication",
    "type": "application/epub+zip"
  }, {
    "href": "http://lsdserver:8990/licenses/15a9b3f9-6ba4-4f11-93cb-5c8ccf3fa336/status",
    "type": "application/vnd.readium.license.status.v1.0+json",
    "rel": "status"
  }, {
    "href": "http://cm.hilbertteam.net:6500/LCP/lcp/hint",
    "rel": "hint"
  }],
```
`The passphrase link` is part of the [LCP Status server's configuration](https://github.com/amigoslibraryservices/lcp-docker/blob/master/base-local/etc/config.yaml.minio#L54).

Reply:
I see -- the LCP server is configured to know about these URLs.